scripts/docker_run

```diff
@@ -3,24 +3,31 @@
 set -o errexit
 set -o xtrace
 
-readonly USERNAME="$(id -u -n)"
+# The default user for a Docker container has uid 0 (root). To avoid creating
+# root-owned files in the build directory we tell Docker to use the current user
+# ID, if known.
+# See
+# https://github.com/googleapis/google-cloud-cpp/blob/a186208b79d900b4ec71c6f9df3acf7638f01dc6/ci/kokoro/docker/build.sh#L147-L152
+readonly DOCKER_UID="${UID:-0}"
+readonly DOCKER_GID="${GID:-0}"
+readonly DOCKER_USER="${USER:-root}"
 
 mkdir -p './bazel-cache'
 mkdir -p './cargo-cache'
 
 docker build \
   --tag=oak \
-  --build-arg="USERNAME=$USERNAME" \
-  --build-arg="UID=$(id -u)" \
-  --build-arg="GID=$(id -g)" \
   .
 
+# Keep this in sync with /scripts/run_examples .
```

Comment:
👍

Reply:
Can we actually move this to a common place instead of relying on a comment?

Reply:
We discussed a flag like `docker_run --detach`. It would be awesome to have the script in that case return the container id as returned by `docker run --detach`.

Reply:
I refactored accordingly: https://github.com/project-oak/oak/pull/112